### DISCUSSION TITLE
Endpoints making use of query parameters & path parameters

### DISCUSSION THREAD

**Discussion by MarkOSullivan94** [serverpod/serverpod#1356] [Q&A] [answered]
I've taken a look at the documentation provided on the [working with endpoints](https://docs.serverpod.dev/concepts/working-with-endpoints) page and also at the examples provided and was unable to find examples of how both query parameters and path parameters are handled

How would I go about implementing both of the endpoints below?

**Query parameter**

`GET https://fairbnb.com/apartments?country=usa&price-limit-upper=2000&bedroom=3&duration=monthly`

Allowing me to pass through multiple optional query parameters which then would be used as filters when querying the DB

**Path parameter**

`GET https://fairbnb.com/apartments/33456204`

Which would then be used when querying the DB to get back a single record

I appreciate any guidance!

**Comment by Frank1016** [comment, marked as answer]
After creating Apartments class in "protocol" folder, you will find the generated "apartments.dart" file in "generated" folder, and this dart file contains all the function calls you can use on Apartments class, including "find" and "findSingleRow". In your endpoint file of this class you can create a query function, using all necessary query parameters in the "where" property of the generated "find" or "findSingleRow". It's similar for the path parmeter, and much eaier.
